domain/schema/model/sql/0015-charm.sql

```diff
@@ -362,15 +362,6 @@ CREATE TABLE charm_tag (
 CREATE INDEX idx_charm_tag_charm
 ON charm_tag (charm_uuid);
 
-CREATE TABLE charm_storage_kind (
-    id INT PRIMARY KEY,
-    name TEXT NOT NULL
-);
-
-INSERT INTO charm_storage_kind VALUES
-(0, 'block'),
-(1, 'filesystem');
```

Comment:
Be careful the tail doesn't wag the dog here i.e. if for some reason we have a new storage kind that isn't available to the charm metadata, that can bite us.

Otherwise, I'm fine with these changes if @manadart is.

Reply:
Storage support in juju is supposed to reflect that described by charm metadata. ie charm metadata is the authorative source for what storage juju can provision.

Reply:
I think we should keep the "charm_" suffixed form and drop the other one; rationale being, that the linkage is all on the charm schema side. We infer it from FKs to `charm_storage`.

Reply:
righto